eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/FixImagesVarDirCommand.php

```diff
@@ -0,0 +1,416 @@
+database references to the new path
+
+Since this script can potentially run for a very long time, to avoid memory exhaustion run it in
+production environment using <info>--env=prod</info> switch.
+
+EOT
+            );
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        /* @var \Doctrine\DBAL\Connection $databaseHandler */
+        $this->connection = $this->getContainer()->get('ezpublish.api.search_engine.legacy.connection');
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|void|null
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $iterationCount = (int) $input->getOption('iteration-count');
+        $this->dryRun = $input->getOption('dry-run');
+        $this->moveFiles = $input->getOption('move-files');
+        $consoleScript = $_SERVER['argv'][0];
+
+        $siteAccess = $this->getContainer()->get('ezpublish.siteaccess');
+
+        /** @var ConfigResolver $configResolver */
+        $configResolver = $this->getContainer()->get('ezpublish.config.resolver');
+        $this->varDir = $configResolver->getParameter(
+            'var_dir',
+            null,
+            $siteAccess->name
+        );
+
+        if (getenv('INNER_CALL')) {
+            $this->processImages($iterationCount, $output);
+            $output->writeln($this->done);
+        } else {
+            $output->writeln([
+                sprintf('Fixing image references using siteaccess %s (var_dir: %s)', $siteAccess->name, $this->varDir),
+                'Calculating number of Images to fix...',
+            ]);
+
+            $count = $this->countImagesToFix();
+            $output->writeln([
+                sprintf('Found total of Images for fixing: %d', $count),
+                '',
+            ]);
+
+            if ($count == 0) {
+                $output->writeln('Nothing to process, exiting.');
+
+                return;
+            }
+
+            $helper = $this->getHelper('question');
+            $question = new ConfirmationQuestion(
+                '<question>Are you sure you want to proceed?</question> ',
+                false
+            );
+
+            if (!$helper->ask($input, $output, $question)) {
+                $output->writeln('');
+
+                return;
+            }
+
+            $progressBar = $this->getProgressBar($count, $output);
+            $progressBar->start();
+
+            for ($fixed = 0; $fixed < $count; $fixed += $iterationCount) {
+                $processScriptFragments = [
+                    $this->getPhpPath(),
+                    $consoleScript,
+                    $this->getName(),
+                    '--iteration-count=' . $iterationCount,
+                ];
+
+                $process = new Process(
```

Comment:
So, with this, re-index command and probably also others, maybe we can find a way to expose reusable code for parallel processes? So one place where we can make sure any global arguments like siteaccess, debug, env, memory, ... is passed to php & bin/console.

Same goes for installers and need to run sub commands, like in [Commerce installer PR](https://github.com/ezsystems/ezcommerce/pull/9).

WDYT @adamwojs @alongosz ?

Reply:
> So side comment, but with this, re-index command and probably also others, maybe we can find a way to expose reusable code for parallel processes? So one place where we can make sure any global arguments like siteaccess, debug, env, memory, ... is passed to php & bin/console.
> 
> Same goes for installers and need to run sub commands, like in [Commerce installer PR](https://github.com/ezsystems/ezcommerce/pull/9).

Yeah, this seems to be very repetitive.